insights-client.spec

```diff
@@ -67,6 +77,16 @@ if [ -d %{_sysconfdir}/motd.d ]; then
     fi
 fi
 
+%post ros
+echo
+echo "Removing custom PCP configuration required for Resource Optimization service!"
+echo
+rm -f /var/lib/pcp/config/pmlogger/config.ros
+sed -i "/PCP_LOG_DIR\/pmlogger\/ros/d" /etc/pcp/pmlogger/control.d/local
+echo
+echo "Enabling ros_collect to send PCP archives for Resource Optimization service!"
+sed -i "s/#ros_collect=True/ros_collect=True/" %{_sysconfdir}/insights-client/insights-client.conf
```

Comment:
Since the config file is a use-editable file, care needs to be taken when automatically editing it. If a diligent user read about this new config file option and before updating their RPM, they add "ros_collect=False" to their configuration file. When this scriptlet runs, this sed line will fail; the file `/etc/insights-client/insights-client.conf` already exists on the user's system from the previously installed RPM. Because this file is a `%config` file, it will not be overwritten when the RPM is upgraded. This scriptlet will not successfully enable the `ros_collect` value because the string "`#ros_collect=True`" is not present in the file.

Reply:
If the default in insights-core will be ros_collect=False, we won't need to run `` may not need to run `sed` at all but just to add this to the original/template (I don't know what the right term is) insights-client.conf file and users will later add it to set it to true if they so want.
@subpop what's your advice?

Reply:
I think the approach that works best for most tooling like this is to append the desired value to the file, flanked by marker lines:

```suggestion
echo >> /etc/insights-client/insights-client.conf <EOF
### Begin insights-client-ros ###
ros_collect=True
### End insights-client-ros ###
EOF
```

It's more wordy, but it explicitly informs the user where this automatic editing of their configuration files came from. It also allows for more targeted removal, should you wish to delete the line upon removal.

```
sed -i 's/### Begin insights-client-ros ###\nros_collect=True\n### End insights-client-ros ###/d' /etc/insights-client/insights-client.conf
```

I think this might handle cases where the user had manually modified the value on their own too. You could also consider adding some conditionality to the scriptlet, where it will only make this modification if the file does not already contain a line beginning with `ros_collect`.

Reply:
Makes sense!

Reply:
Okay I can see below scenarios,

1. User already enabled the ros with ros_collect=True . In this case we don't need anything to happen as he already opted in for that.
2. User already disabled the ros with ros_collect=False. In this case as well we should not enable it forcefully.
3. It can be True or False however commented. We are only enabling it if its #ros_collect=True.

The current sed takes care of above scenarios. Considering your suggestion to make it more wordy if I understand it means,

1. Remove the default commented config from insights-client.conf
2. Add the param if its(not commented)  not there.
3. What should be correct action if its commented and either True or False? I think we should not update it and enable in this situation? because in current sed we enable it if its #ros_collect=True

Reply:
> Remove the default commented config from insights-client.conf

I think that this configuration flag is obscure enough that it does not warrant a commented out default value. We don't have anything related to malware detection or compliance in the default values/comments in insights-client.conf, so I don't think that ROS should be there either.

> Add the param if its(not commented) not there.

Typically, when automatically modifying configuration files that are also expected to be edited manually by users, great care should be taken not to overwrite or otherwise overrule the user modifications. If an uncommented line beginning with `ros_collect` is already present in the file, no automatic action should take place; the user has clearly already indicated their preference through their explicit action editing the file. So your scriptlet should only take action when no explicit value is already set in the file.

> What should be correct action if its commented and either True or False? I think we should not update it and enable in this situation? because in current sed we enable it if its #ros_collect=True and only not handling #ros_collect=False situation.

If the default state of the configuration file is to *not* have this value present, the presence of a commented "ros_collect" line again means user interaction; ignore it.

This is why using a sentinel-marked comment block makes for a clear indication of what a script is doing to a configuration file. Your scriptlets need only interact with the lines between (inclusively) your sentinel comments.

Reply:
Thanks for explanations. So  request to review latest commit which has the modifications accordingly. Thank you!